src/logic/calculate.js

```diff
@@ -0,0 +1,16 @@
+import operate from './operate';
+
+const calculate = ({ total, next }, bName) => {
+  let result = total;
+  let nextNum = next;
+
+  if (bName === '+/-') {
+    result *= -1;
+    nextNum *= -1;
+    return { result, nextNum };
+  }
+  result = operate(result, nextNum, bName);
+  return { result, nextNum };
+};
```

Comment:
The `calculator data object` should contain three properties: `total`, `next` and `operation` ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/react_reducx/calculator/milestone4/totalNextOperate.png)). You seem to have ignored the 3rd property in your logic.